guides/TrainingModelsHPC.md

```diff
@@ -92,76 +92,113 @@
 >
 > If we launch a job and then modify the config file _before_ the job has been able to read it, we may be using an undesired version of the config in our job! To avoid this, it is best to wait until you can verify in MLflow that the job has the expected config parameters (and then edit the file to launch a new job if needed).
 
-6.  **Run the training job using the SLURM scheduler**
+6. **Optional argument - Optuna**
 
-    To launch a job, use the `sbatch` command with the relevant training script:
+   - We have the option to run [Optuna](https://optuna.org) which is a hyperparameter optimization framework. This enables automated hyperparameter optimization by selecting the best parameters based on a chosen optimizer.
+   - Currently, Optuna can be used to find the best number of epochs and learning rate within specified ranges for specific number of trials.
+   - Specify the range of values for each hyperparameter you wish to optimize in the configuration file (for eaxample like in `/crabs/detection_tracking/config/faster_rcnn.yaml`) under `optuna_param`.
+   - Determine how many trials you want Optuna to run. Each trial will test a different combination of hyperparameters within the defined search space.
+   - Optuna will run the specified number of trials, comparing the metrics for each trial to determine the best set of hyperparameters.
+   - After `n trials`, MLflow will log the best hyperparameters identified by Optuna. Subsequently, the model will be trained using these optimized parameters.
 
-    ```
-    sbatch <path-to-training-bash-script>
-    ```
+   To run Optuna, we can add `--optuna` arguments to the bash script. For example:
```

Comment:
Thank you for this. I have added a bit more clarification. Technically optuna is not hyperparameter sweep. It is a hyperparameter  optimisation, as it apply some optimization algorithms rather than random search.